code/modules/unit_tests/unit_test.dm

```diff
@@ -78,49 +80,59 @@ GLOBAL_VAR(test_log)
 	allocated += instance
 	return instance
 
-/proc/RunUnitTests()
-	CHECK_TICK
+/proc/RunUnitTest(test_path, list/test_results)
+	var/datum/unit_test/test = new test_path
 
-	var/tests_to_run = subtypesof(/datum/unit_test)
-	for (var/_test_to_run in tests_to_run)
-		var/datum/unit_test/test_to_run = _test_to_run
-		if (initial(test_to_run.focus))
-			tests_to_run = list(test_to_run)
-			break
+	GLOB.current_test = test
+	var/duration = REALTIMEOFDAY
 
-	var/list/test_results = list()
+	test.Run()
 
-	for(var/I in tests_to_run)
-		var/datum/unit_test/test = new I
+	duration = REALTIMEOFDAY - duration
+	GLOB.current_test = null
+	GLOB.failed_any_test |= !test.succeeded
 
-		GLOB.current_test = test
-		var/duration = REALTIMEOFDAY
+	var/list/log_entry = list("[test.succeeded ? "PASS" : "FAIL"]: [test_path] [duration / 10]s")
+	var/list/fail_reasons = test.fail_reasons
 
-		test.Run()
+	for(var/J in 1 to LAZYLEN(fail_reasons))
+		log_entry += "\tREASON #[J]: [fail_reasons[J]]"
+	var/message = log_entry.Join("\n")
+	log_test(message)
 
-		duration = REALTIMEOFDAY - duration
-		GLOB.current_test = null
-		GLOB.failed_any_test |= !test.succeeded
+	test_results[test_path] = list("status" = test.succeeded ? UNIT_TEST_PASSED : UNIT_TEST_FAILED, "message" = message, "name" = test_path)
 
-		var/list/log_entry = list("[test.succeeded ? "PASS" : "FAIL"]: [I] [duration / 10]s")
-		var/list/fail_reasons = test.fail_reasons
+	qdel(test)
 
-		for(var/J in 1 to LAZYLEN(fail_reasons))
-			log_entry += "\tREASON #[J]: [fail_reasons[J]]"
-		var/message = log_entry.Join("\n")
-		log_test(message)
+/proc/RunUnitTests()
+	CHECK_TICK
 
-		test_results[I] = list("status" = test.succeeded ? UNIT_TEST_PASSED : UNIT_TEST_FAILED, "message" = message, "name" = I)
+	var/list/tests_to_run = subtypesof(/datum/unit_test)
+	for (var/_test_to_run in tests_to_run)
+		var/datum/unit_test/test_to_run = _test_to_run
+		if(initial(test_to_run.snowflake))
+			tests_to_run -= test_to_run
+			continue
+		if (initial(test_to_run.focus))
+			tests_to_run = list(test_to_run)
+			break
 
-		qdel(test)
+	var/list/test_results = list()
 
+	for(var/unit_path in tests_to_run)
+		RunUnitTest(unit_path, test_results)
 		CHECK_TICK
 
+	//Fuck you, fight me
+	RunUnitTest(/datum/unit_test/del_test, test_results)
```

Comment:
This is dumb, just have a priority var that you sort a list of tests by.

Reply:
sure